src/schema/v1/show.ts

```diff
@@ -664,16 +665,37 @@ const Show: GraphQLFieldConfig<void, ResolverContext> = {
       description: "The slug or ID of the Show",
     },
   },
-  resolve: (_root, { id }, { showLoader }) => {
-    return showLoader(id)
+  resolve: (
+    _root,
+    { id },
+    {
+      unauthenticatedLoaders: { showLoader: loaderWithCache },
+      authenticatedLoaders: { showLoader: loaderWithoutCache },
+      accessToken,
+    }
+  ) => {
+    const decodeUnverifiedJwt = decodeUnverifiedJWT(accessToken as string)
+    const partnerIds: Array<String> = decodeUnverifiedJwt
+      ? decodeUnverifiedJwt.partner_ids
+      : []
+    const isAdmin: boolean =
+      decodeUnverifiedJwt &&
+      decodeUnverifiedJwt.roles.split(",").includes("admin")
+    const isDisplayable = show =>
+      show.displayable || isAdmin || partnerIds.includes(show.partner._id)
+
+    const loader =
+      decodeUnverifiedJwt && loaderWithoutCache
+        ? loaderWithoutCache
+        : loaderWithCache
```

Comment:
If I understand correctly, we don't need to differentiate `loaderWithCache` and `loaderWithoutCache` here because they are both hitting the same Gravity endpoint (`/show/:id`) the same way. Since we name the loaders the same, depending if there is `accessToken` in the request, we will automatically use authenticated/non-authenticated loader and get desired caching behavior.

I could be totally wrong here. cc @mzikherman